crates/cosmos/cosmos-chain-components/src/impls/queries/write_ack_event.rs

```diff
@@ -1,65 +1,88 @@
+use crate::traits::grpc_address::HasGrpcAddress;
+use crate::traits::rpc_client::HasRpcClient;
+use cgp_core::prelude::*;
 use hermes_relayer_components::chain::traits::queries::write_ack::WriteAckQuerier;
 use hermes_relayer_components::chain::traits::types::ibc_events::write_ack::HasWriteAckEvent;
 use hermes_relayer_components::chain::traits::types::packet::HasIbcPacketTypes;
-use ibc_relayer::chain::handle::ChainHandle;
+use ibc_relayer::chain::cosmos::query::tx::query_packets_from_block;
 use ibc_relayer::chain::requests::Qualified;
-use ibc_relayer::link::packet_events::query_write_ack_events;
-use ibc_relayer::path::PathIdentifiers;
+use ibc_relayer::chain::requests::QueryHeight;
+use ibc_relayer::chain::requests::QueryPacketEventDataRequest;
+use ibc_relayer_types::core::ics02_client::error::Error as Ics02Error;
+use ibc_relayer_types::core::ics04_channel::error::Error as Ics04Error;
 use ibc_relayer_types::core::ics04_channel::events::WriteAcknowledgement;
 use ibc_relayer_types::core::ics04_channel::packet::Packet;
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
 use ibc_relayer_types::events::IbcEvent;
+use ibc_relayer_types::events::WithBlockDataType;
+use ibc_relayer_types::Height;
+use tendermint_rpc::{Client, Error as TendermintRpcError};
 
-use crate::traits::chain_handle::HasBlockingChainHandle;
+pub struct QueryWriteAckEventFromAbci;
 
-pub struct QueryWriteAckEventFromChainHandle;
-
-impl<Chain, Counterparty> WriteAckQuerier<Chain, Counterparty> for QueryWriteAckEventFromChainHandle
+impl<Chain, Counterparty> WriteAckQuerier<Chain, Counterparty> for QueryWriteAckEventFromAbci
 where
-    Chain: HasWriteAckEvent<Counterparty, WriteAckEvent = WriteAcknowledgement>
+    Chain: HasErrorType
+        + HasWriteAckEvent<Counterparty, WriteAckEvent = WriteAcknowledgement>
         + HasIbcPacketTypes<Counterparty, IncomingPacket = Packet>
-        + HasBlockingChainHandle,
+        + HasRpcClient
+        + HasGrpcAddress
+        + HasComponents
+        + CanRaiseError<TendermintRpcError>
+        + CanRaiseError<Ics04Error>
+        + CanRaiseError<Ics02Error>
+        + CanRaiseError<tonic::transport::Error>
+        + CanRaiseError<ibc_relayer::error::Error>,
```

Comment:
The following constraints can be safely removed without compilation error:

- `HasErrorType` (because we have `CanRaiseError` which includes it as a supertrait)
- `HasGrpcAddress`
- `HasComponents`
- `CanRaiseError<Ics04Error>`
- `CanRaiseError<tonic::transport::Error>`

In general, a constraint can be removed if it does not introduce any compile error.